src/setup_payload/AdditionalDataPayloadGenerator.cpp

```diff
@@ -53,7 +53,7 @@ AdditionalDataPayloadGenerator::generateAdditionalDataPayload(uint16_t lifetimeC
     size_t rotatingDeviceIdBufferSize = 0;
 
     // Initialize TLVWriter
-    writer.Init(chip::System::PacketBufferHandle::New(chip::System::PacketBuffer::kMaxSize));
+    writer.Init(chip::System::PacketBufferHandle::New(RotatingDeviceId::kRotatingDeviceIdMaxLength));
 
     SuccessOrExit(err = writer.OpenContainer(AnonymousTag, kTLVType_Structure, innerWriter));
 
```

Comment:
The code below has to be conditional on Rotating Device ID being present/enabled.

Suggest both enabling and disabling rotating ID in the build today, locally, to see if there are bugs.

Reply:
i did enable/disable the rotating id flags to make sure there is no bugs
this is just a utility for generating the rotating id, i think the actual core logic should have the flags (which they do now)